src/main/java/io/confluent/connect/hdfs/HdfsSinkConnectorConfig.java

```diff
@@ -330,10 +332,37 @@ private void addConfig(Map<String, ?> parsedProps, ComposableConfig config) {
     }
   }
 
+  /**
+   * Returns the url property. Preference is given to property <code>store.url</code> over
+   * <code>hdfs.url</code> because <code>hdfs.url</code> is deprecated.
+   *
+   * @param props - Map of properties for the connector
+   * @return String - url for HDFS
+   */
+  private String extractUrl(Map<String, String> props) {
+    String storageUrl = props.get(StorageCommonConfig.STORE_URL_CONFIG);
+    if (storageUrl != null && !storageUrl.equals(StorageCommonConfig.STORE_URL_DEFAULT)) {
+      return storageUrl;
+    }
+
+    String hdfsUrl = props.get(HDFS_URL_CONFIG);
+    if (hdfsUrl != null && !hdfsUrl.equals(HDFS_URL_DEFAULT)) {
+      return hdfsUrl;
+    }
+
+    throw new ConfigException(
+        String.format("Configuration %s cannot be empty."), StorageCommonConfig.STORE_URL_CONFIG
+    );
+  }
+
   public String getName() {
     return name;
   }
 
+  public String getUrl() {
+    return url;
```

Comment:
Given that this is called in `DataWriter` anyways, I believe we can just keep the logic in here and not introduce `extractUrl`

Reply:
considering it gets called multiple places I think it's fine to preprocess

Reply:
It's good to keep in mind that it's not in a critical path and `url` is not used heavily. So reducing boiler place and number of declared methods could also be seen as an optimization. Not a blocker though